### DISCUSSION TITLE
Nixifying https://github.com/haskell/cabal using haskell-flake

### DISCUSSION THREAD

**Discussion by ParetoOptimalDev** [srid/haskell-flake#80] [General]
Split off from #76 as not to go off topic.

I tried this and it doesn't work yet, but I'd like for it to work eventually. As a stopgap though I did submit a PR with just a `haskell-parts` flake at https://github.com/haskell/cabal/pull/8753#issuecomment-1426339239 so that contributing from nix can at least be done with a simple `cabal build cabal` in https://github.com/haskell/cabal/pull/8753.

However even that might not be merged given there are concerns about maintaing that simple of a flake... so I guess this is a thread to record my intent, foster discussion, and provide guideposts for anyone in the future who may want this.

**Comment by srid** [comment]
> there are concerns about maintaing that simple of a flake.

I've always found a large portion of open source community (I'm mainly familiar with Haskell projects, fwiw) to be at best indifferent to Nix if not somehow ill-disposed. That's something to keep in mind as you spend time & effort contributing Nix support in those projects. 

To be fair, Nix is still a bit of a niche, and it doesn't yet have great documentation for newcomers - so right now, IMO, it tends to attract [geeks](https://meaningness.com/geeks-mops-sociopaths) more than anyone else. Identifying the subset of project maintainers that are _open_ to adding (and learning from) Nix support could be an interesting exercise.

**Reply by srid** [reply]
> Identifying the subset of project maintainers that are _open_ to adding (and learning from) Nix support could be an interesting exercise.

Example: https://github.com/soywod/himalaya/pull/76

**Reply by ParetoOptimalDev** [reply]
> Identifying the subset of project maintainers that are open to adding (and learning from) Nix support could be an interesting exercise.

That sounds like a very fruitful endeavor that makes both sides happy!
